subgraphs/morpho/src/helpers.ts

```diff
@@ -75,99 +81,159 @@ export function createInterestRate(
-    marketDailySnapshot.dailyProtocolSideRevenueUSD.plus(newProtocolRevenue);
+    marketDailySnapshot.dailyProtocolSideRevenueUSD.plus(protocolSideRevenue);
+  marketHourlySnapshot.cumulativeProtocolSideRevenueUSD =
+    market.cumulativeProtocolSideRevenueUSD;
+  marketHourlySnapshot.hourlyProtocolSideRevenueUSD =
+    marketHourlySnapshot.hourlyProtocolSideRevenueUSD.plus(protocolSideRevenue);
+
+  market.cumulativeTotalRevenueUSD =
+    market.cumulativeTotalRevenueUSD.plus(totalRevenue);
+  marketDailySnapshot.cumulativeTotalRevenueUSD =
+    market.cumulativeTotalRevenueUSD;
   marketDailySnapshot.dailyTotalRevenueUSD =
-    marketDailySnapshot.dailyTotalRevenueUSD.plus(newTotalRevenue);
-  marketDailySnapshot.save();
+    marketDailySnapshot.dailyTotalRevenueUSD.plus(totalRevenue);
+  marketHourlySnapshot.cumulativeTotalRevenueUSD =
+    market.cumulativeTotalRevenueUSD;
+  marketHourlySnapshot.hourlyTotalRevenueUSD =
+    marketHourlySnapshot.hourlyTotalRevenueUSD.plus(totalRevenue);
 
-  // get and update market daily snapshot
-  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(
-    market,
-    timestamp,
-    blockNumber
+  marketDailySnapshot.rates = getSnapshotRates(
+    market.rates,
+    (block.timestamp.toI32() / SECONDS_PER_DAY).toString()
   );
 
-  // update hourly revenues
-  marketHourlySnapshot.hourlySupplySideRevenueUSD =
-    marketHourlySnapshot.hourlySupplySideRevenueUSD.plus(newSupplyRevenue);
-  marketHourlySnapshot.hourlyProtocolSideRevenueUSD =
-    marketHourlySnapshot.hourlyProtocolSideRevenueUSD.plus(newProtocolRevenue);
-  marketHourlySnapshot.hourlyTotalRevenueUSD =
-    marketHourlySnapshot.hourlyTotalRevenueUSD.plus(newTotalRevenue);
   marketHourlySnapshot.save();
+  marketDailySnapshot.save();
+  financialSnapshot.save();
+  protocol.save();
+  market.save();
+}
+
+export function getOrCreateFinancialDailySnapshots(
+  protocol: LendingProtocol,
+  block: ethereum.Block
+): FinancialsDailySnapshot {
+  const snapshotId = getDayId(block.timestamp);
+  let financialMetrics = FinancialsDailySnapshot.load(snapshotId);
+
+  if (!financialMetrics) {
+    financialMetrics = new FinancialsDailySnapshot(snapshotId);
+    financialMetrics.protocol = protocol.id;
+
+    financialMetrics.days = getDay(block.timestamp).toI32();
+    financialMetrics.dailyDepositUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyBorrowUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyLiquidateUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyWithdrawUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyRepayUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailySupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyTotalRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyTransferUSD = BIGDECIMAL_ZERO;
+    financialMetrics.dailyFlashloanUSD = BIGDECIMAL_ZERO;
+
+    financialMetrics.totalValueLockedUSD = BIGDECIMAL_ZERO;
```

Comment:
I'm not a huge fan of the pattern of initializing everything to 0. It makes it easier to forget to update the fields after the snapshot is created. I would prefer we take the `protocol` entity and set any fields that we can to the `protocol` fields.

This was actually causing issues in some YAs

Reply:
I see. I will update this.